### PR TITLE
[Merged by Bors] - feat: intervals with rational bounds generate the real Borel sigma algebra

### PR DESCRIPTION
**Pull request by RemyDegenne**
Add `borel_eq_generateFrom_Ioi_rat`, `borel_eq_generateFrom_Ici_rat`, `borel_eq_generateFrom_Iic_rat` (we already have the `Iio` result). Also prove that these sets of intervals are pi-systems.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/MeasureTheory/Constructions/BorelSpace/Basic.lean [hunk 1]
```diff
@@ -1697,13 +1697,105 @@ theorem borel_eq_generateFrom_Ioo_rat :
...
+    refine generateFrom_le fun t ht => ?_
+    simp only [iUnion_singleton_eq_range, mem_range] at ht
+    obtain ⟨y, rfl⟩ := ht
+    rw [← compl_Iic]
+    refine MeasurableSet.compl (GenerateMeasurable.basic _ ?_)
+    simp only [iUnion_singleton_eq_range, mem_range, exists_apply_eq_apply]
+  · refine MeasurableSpace.generateFrom_le fun s hs => ?_
+    simp only [iUnion_singleton_eq_range, mem_range] at hs
+    obtain ⟨y, rfl⟩ := hs
+    exact measurableSet_Iic
```
Comment (ADedecker) [inline Mathlib/MeasureTheory/Constructions/BorelSpace/Basic.lean:1756]:
```suggestion
  rw [borel_eq_generateFrom_Ioi_rat, iUnion_singleton_eq_range, iUnion_singleton_eq_range]
  refine' le_antisymm (generateFrom_le _) (generateFrom_le _) <;>
  rintro _ ⟨q, rfl⟩ <;>
  dsimp only <;>
  [rw [← compl_Iic]; rw [← compl_Ioi]] <;>
  exact MeasurableSet.compl (GenerateMeasurable.basic _ (mem_range_self q))
```

File: Mathlib/MeasureTheory/Constructions/BorelSpace/Basic.lean [hunk 2]
```diff
@@ -1697,13 +1697,105 @@ theorem borel_eq_generateFrom_Ioo_rat :
...
+    exact measurableSet_Ici
+
 theorem isPiSystem_Ioo_rat :
     @IsPiSystem ℝ (⋃ (a : ℚ) (b : ℚ) (_ : a < b), {Ioo (a : ℝ) (b : ℝ)}) := by
   convert isPiSystem_Ioo ((↑) : ℚ → ℝ) ((↑) : ℚ → ℝ)
   ext x
   simp [eq_comm]
 #align real.is_pi_system_Ioo_rat Real.isPiSystem_Ioo_rat
 
+theorem isPiSystem_Iio_rat : @IsPiSystem ℝ (⋃ a : ℚ, {Iio (a : ℝ)}) := by
```
Comment (ADedecker) [inline Mathlib/MeasureTheory/Constructions/BorelSpace/Basic.lean:1779]: Why do you need the `@`s here?

**Comment by ADedecker** [comment]
Thanks! 

bors r+

**Comment by RemyDegenne** [comment]
bors r+
I think it's ok to do that myself since the only commit since the last bors r+ is a trivial fix.

**Pull request closed without merging**